stats/record.go

```diff
@@ -52,3 +52,17 @@ func Record(ctx context.Context, ms ...Measurement) {
 	}
 	recorder(tag.FromContext(ctx), ms)
 }
+
+// RecordWithTags records one or multiple measurements at once.
+//
+// Measurements will be tagged with the tags in the context mutated by the mutators.
+// RecordWithTags is useful if you want to record with tag mutations but don't want
+// to propagate the mutations in the context.
+func RecordWithTags(ctx context.Context, mutators []tag.Mutator, ms ...Measurement) error {
```

Comment:
I don't really mind but I think in this case it isn't worth having varargs for the `ms` argument. I think since we are requiring the user to explicitly pass a `tag.Mutator` slice, we might as well also require an explicit `ms` slice. 

The small advantage is that we could in future introduce a new varargs argument if we ever found the need for that in a backwards-compatible way. 

This is just a thought, I don't really mind if we leave it like this either.

Reply:
I like it as it is because the APIs are more symmetric.